examples/adwords_axis/src/main/java/adwords/axis/v201802/campaignmanagement/AddCompleteCampaignsUsingBatchJob.java

```diff
@@ -0,0 +1,451 @@
+        biddableAdGroupCriterion.setCriterion(keyword);
+
+        // Create AdGroupCriterionOperation.
+        AdGroupCriterionOperation operation = new AdGroupCriterionOperation();
+        operation.setOperand(biddableAdGroupCriterion);
+        operation.setOperator(Operator.ADD);
+
+        // Add to list.
+        adGroupCriteriaOperations.add(operation);
+      }
+    }
+    return adGroupCriteriaOperations;
+  }
+
+  private static List<AdGroupOperation> buildAdGroupOperations(Iterator<Long> tempIdGenerator,
+      String namePrefix, Iterable<CampaignOperation> campaignOperations) {
+    List<AdGroupOperation> operations = new ArrayList<>();
+    for (CampaignOperation campaignOperation : campaignOperations) {
+      for (int i = 0; i < NUMBER_OF_ADGROUPS_TO_ADD; i++) {
+        AdGroup adGroup = new AdGroup();
+        adGroup.setCampaignId(campaignOperation.getOperand().getId());
+        adGroup.setId(tempIdGenerator.next());
+        adGroup.setName(String.format("Batch Ad Group %s.%s", namePrefix, i));
+
+        BiddingStrategyConfiguration biddingStrategyConfiguration =
+            new BiddingStrategyConfiguration();
+        CpcBid bid = new CpcBid();
+        bid.setBid(new Money(null, 10000000L));
+        biddingStrategyConfiguration.setBids(new Bids[] {bid});
+
+        adGroup.setBiddingStrategyConfiguration(biddingStrategyConfiguration);
+
+        AdGroupOperation operation = new AdGroupOperation();
+        operation.setOperand(adGroup);
+        operation.setOperator(Operator.ADD);
+
+        operations.add(operation);
+      }
+    }
+    return operations;
+  }
+
+  private static List<CampaignCriterionOperation> buildCampaignCriterionOperations(
+      List<CampaignOperation> campaignOperations) {
+
+    List<CampaignCriterionOperation> operations = new ArrayList<>();
+    for (CampaignOperation campaignOperation : campaignOperations) {
+      Keyword keyword = new Keyword();
+      keyword.setMatchType(KeywordMatchType.BROAD);
+      keyword.setText("venus");
+
+      NegativeCampaignCriterion negativeCriterion = new NegativeCampaignCriterion();
+      negativeCriterion.setCampaignId(campaignOperation.getOperand().getId());
+      negativeCriterion.setCriterion(keyword);
+
+      CampaignCriterionOperation operation = new CampaignCriterionOperation();
+      operation.setOperand(negativeCriterion);
+      operation.setOperator(Operator.ADD);
+
+      operations.add(operation);
+    }
+
+    return operations;
+  }
+
+  private static List<CampaignOperation> buildCampaignOperations(
+      Iterator<Long> tempIdGenerator, String namePrefix, BudgetOperation budgetOperation) {
+    long budgetId = budgetOperation.getOperand().getBudgetId();
+
+    List<CampaignOperation> operations = new ArrayList<>();
+    for (int i = 0; i < NUMBER_OF_CAMPAIGNS_TO_ADD; i++) {
+      Campaign campaign = new Campaign();
+      campaign.setName(String.format("Batch Campaign %s.%s", namePrefix, i));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: i (declared at line 404). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 406. Consider formatting this data with &#39;%d&#39; instead.